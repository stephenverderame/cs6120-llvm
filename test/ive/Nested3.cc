// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    for (int i = 0; i < 10; ++i) {
        int p = m + i * 10;
        for (int j = 0; j < i; ++j) {
            int k = j + i * p;
            for (int m = 0; m < k; m += 20) {
                int n = m + k * p;
                printf("%d\n", n);
            }

            for (int m = 0; m < k; m++) {
                int n = k - m * p;
                printf("%d\n", n);
            }
        }

        for (int j = 0; j < i; ++j) {
            int k = j - i * p;
            printf("%d\n", k);
        }
    }
}
// NOLINTEND