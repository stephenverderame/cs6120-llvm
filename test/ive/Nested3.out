Loop: 
13:                                               ; preds = %20, %10
  %.03 = phi i32 [ 0, %10 ], [ %21, %20 ]
  %14 = icmp slt i32 %.03, %12
  br i1 %14, label %16, label %15

Basic IV:   %.03 = phi i32 [ 0, %10 ], [ %21, %20 ]
Basic IV:   %21 = add nsw i32 %.03, 20
Loop: 
23:                                               ; preds = %30, %22
  %.01 = phi i32 [ 0, %22 ], [ %31, %30 ]
  %24 = icmp slt i32 %.01, %12
  br i1 %24, label %26, label %25

Basic IV:   %.01 = phi i32 [ 0, %22 ], [ %31, %30 ]
Derived IV: %27 = %.01 * %6
Basic IV:   %31 = add nsw i32 %.01, 1
Loop: 
7:                                                ; preds = %33, %4
  %.04 = phi i32 [ 0, %4 ], [ %34, %33 ]
  %8 = icmp slt i32 %.04, %.02
  br i1 %8, label %10, label %9

Basic IV:   %.04 = phi i32 [ 0, %4 ], [ %34, %33 ]
Basic IV:   %34 = add nsw i32 %.04, 1
Loop: 
36:                                               ; preds = %43, %35
  %.0 = phi i32 [ 0, %35 ], [ %44, %43 ]
  %37 = icmp slt i32 %.0, %.02
  br i1 %37, label %39, label %38

Basic IV:   %.0 = phi i32 [ 0, %35 ], [ %44, %43 ]
Basic IV:   %44 = add nsw i32 %.0, 1
Loop: 
1:                                                ; preds = %46, %0
  %.02 = phi i32 [ 0, %0 ], [ %47, %46 ]
  %2 = icmp slt i32 %.02, 10
  br i1 %2, label %4, label %3

Basic IV:   %.02 = phi i32 [ 0, %0 ], [ %47, %46 ]
Basic IV:   %47 = add nsw i32 %.02, 1
Derived IV: %5 = %.02 * 10
Derived IV: %6 = %.02 * 10 + 20
